app/screens/custom_status/custom_status_modal.tsx

```diff
@@ -0,0 +1,391 @@
+        const {customStatus} = this.props;
+        if (isStatusSet) {
+            const isStatusSame = customStatus.emoji === emoji && customStatus.text === text;
+            if (!isStatusSame) {
+                const status = {
+                    emoji: emoji || 'speech_balloon',
+                    text: text.trim(),
+                };
+                this.props.actions.setCustomStatus(status);
+            }
+        } else if (customStatus && customStatus.emoji) {
+            this.props.actions.unsetCustomStatus();
+        }
+        Keyboard.dismiss();
+        dismissModal();
+    };
+
+    handleTextChange = (value: string) => this.setState({text: value});
+
+    handleRecentCustomStatusClear = (status: UserCustomStatus) => this.props.actions.removeRecentCustomStatus(status);
+
+    clearHandle = () => {
+        this.setState({emoji: '', text: ''});
+    };
+
+    handleSuggestionClick = (status: UserCustomStatus) => {
+        const {emoji, text} = status;
+        this.setState({emoji, text});
+    };
+
+    renderRecentCustomStatuses = () => {
+        const {recentCustomStatuses, theme} = this.props;
+        const style = getStyleSheet(theme);
+        const recentStatuses = recentCustomStatuses.map((status: UserCustomStatus, index: number) => (
+            <CustomStatusSuggestion
+                key={status.text}
+                handleSuggestionClick={this.handleSuggestionClick}
+                handleClear={this.handleRecentCustomStatusClear}
+                emoji={status.emoji}
+                text={status.text}
+                theme={theme}
+                separator={index !== recentCustomStatuses.length - 1}
+            />
+        ));
+
+        if (recentStatuses.length <= 0) {
+            return null;
+        }
+
+        return (
+            <>
+                <View style={style.separator}/>
+                <View testID='custom_status.recents'>
+                    <Text style={style.title}>
+                        {this.props.intl.formatMessage({
+                            id: 'custom_status.suggestions.recent_title',
+                            defaultMessage: 'RECENT',
+                        })}
+                    </Text>
+                    <View style={style.block}>
+                        {recentStatuses}
+                    </View>
+                </View >
+            </>
+        );
+    };
+
+    renderCustomStatusSuggestions = () => {
+        const {recentCustomStatuses, theme} = this.props;
+        const style = getStyleSheet(theme);
+        const recentCustomStatusTexts = recentCustomStatuses.map((status: UserCustomStatus) => status.text);
+        const customStatusSuggestions = defaultCustomStatusSuggestions.
+            map((status) => ({
+                emoji: status.emoji,
+                text: status.messageDefault,
+            })).
+            filter((status: UserCustomStatus) => !recentCustomStatusTexts.includes(status.text)).
```

Comment:
interesting what would happen if I change my language after having set a suggestion in a previous language as the text won't match

Reply:
It will not filter out that status. We have implemented it the same as it's implemented in the webapp.

Reply:
Well the implementation on webapp needs fixing in my opinion

Reply:
do we have a ticket for this?
cc: @esethna